src/github.com/couchbase/sync_gateway/db/kv_change_index.go

```diff
@@ -205,6 +184,62 @@ func (k *kvChangeIndex) initIndexPartitions() (*base.IndexPartitions, error) {
 	return k.indexPartitions, nil
 }
 
+func (k *kvChangeIndex) loadIndexPartitionsFromBucket() (base.PartitionStorageSet, error) {
+	var partitionDef base.PartitionStorageSet
+	value, _, err := k.reader.indexReadBucket.GetRaw(base.KIndexPartitionKey)
+	if err == nil {
+		if err = json.Unmarshal(value, &partitionDef); err != nil {
+			return nil, err
+		}
+	}
+	return partitionDef, nil
+}
+
+func (k *kvChangeIndex) retrieveCBGTPartitions() (partitionDef base.PartitionStorageSet, err error) {
+
+	var manager *cbgt.Manager
+	if k.context != nil {
+		manager = k.context.BucketSpec.CbgtContext.Manager
+	} else {
+		return nil, errors.New("Unable to determine partition map for index - not found in index, and no database context")
```

Comment:
Can you add a comment or extend error message to explain implications of this error?  Will it fix itself?